src/Routing/Router.php

```diff
@@ -1015,6 +1015,7 @@ public static function parseRoutePath(string $url): array
             (?<controller>[a-z0-9]+)
             ::
             (?<action>[a-z0-9_]+)
+            (?<params>[a-z0-9-_/=]*)
```

Comment:
```suggestion
            (?<params>(?:/(?:[a-z][a-z0-9-_]*=)?[a-z0-9-_]+)+/?)
```
These won't be accepted anymore:
```
Bookmarks::view-1
Bookmarks::view=1
Bookmarks::view/=value
Bookmarks::view/key=
Bookmarks::view/key=value=value
Bookmarks::view/0key=value
Bookmarks::view/-key=value
```

Reply:
@markstory you want to have tests added for those kind of scenarios?

Reply:
actually this breaks other tests as well so I will revert that commit

Reply:
```suggestion
            (?<params>(?:/(?:[a-z][a-z0-9-_]*=)?[a-z0-9-_]+)+/?)?
```
I've forgot `?` in end to make params optional. I've tested only with params, so I missed cases without params.

Reply:
> you want to have tests added for those kind of scenarios?

I don't think we need to have tests for all the invalid data values. But we should have coverage on the valid states that we expect users to use. I think we should not allow keys with leading numbers as they can't be turned into PHP variables with passed parameters. While they are valid outside of that context, we shouldn't have incompatible features within the framework.

Reply:
Found (another) regex to validate the key string is a valid variable name.

Reply:
The only thing I am struggeling with right now is the `Could not parse a key=value from` exception. Can't find a good example where this exception would throw so I am thinking of just removing that check...